.github/workflows/ci-common.yml

```diff
@@ -0,0 +1,219 @@
+# See reference docs at
+# https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions
+name: Default branch ci
+on:
+  workflow_call:
+    inputs:
+      container-repo:
+        required: true
+        type: string
+      container-version:
+        required: true
+        type: string
+
+jobs:
+  check-sigs:
+    runs-on: ubuntu-22.04
+    container: ${{ inputs.container-repo }}:${{ inputs.container-version }}
+    steps:
+      - name: Clone the repo
+        uses: actions/checkout@v4
+
+      - name: Check sigs
+        run: ./.ci/check-release-sigs
+
+  check-hashes:
+    runs-on: ubuntu-22.04
+    container: ${{ inputs.container-repo }}:${{ inputs.container-version }}
+    steps:
+      - name: Clone the repo
+        uses: actions/checkout@v4
+
+      - name: Check sigs
+        run: ./.ci/check-hashes
+
+  lint-py:
+    runs-on: ubuntu-22.04
+    container: ${{ inputs.container-repo }}:${{ inputs.container-version }}
+    steps:
+      - name: Clone the repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Mark directory as safe
+        run: git config --global --add safe.directory $GITHUB_WORKSPACE
+
+      - name: Lint python (black, mypy, pylint)
+        run: |
+          make -C py clean
+          make -C py
+          ./.ci/check-pep8
+
+  lint-src:
+    runs-on: ubuntu-22.04
+    container:
+      image: ${{ inputs.container-repo }}:${{ inputs.container-version }}
+      env:
+        RUSTFLAGS: "-Dwarnings"
+    steps:
+      - name: Clone the repo
+        uses: actions/checkout@v4
+        with:
+          # Unfortunately linting the sources requires running cmake, so we need to do a full checkout
+          fetch-depth: 0
+          fetch-tags: true
+          submodules: recursive
+
+      - name: Mark directory as safe
+        run: git config --global --add safe.directory $GITHUB_WORKSPACE
+
+      - name: Lint src code style (clang-format, rustfmt)
+        run: ./.ci/check-style
+
+      - name: Lint src bugs/best-practice (clang-tidy, clippy)
+        run: |
+          make prepare-tidy
+          ./.ci/check-tidy
+          make run-rust-clippy
+
+  unit-tests:
+    runs-on: ubuntu-22.04
+    container:
+      image: ${{ inputs.container-repo }}:${{ inputs.container-version }}
+      env:
+        RUSTFLAGS: "-Dwarnings"
+    steps:
+      - name: Clone the repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+          submodules: recursive
+
+      - name: Mark directory as safe
+        run: git config --global --add safe.directory $GITHUB_WORKSPACE
+
+      - name: Build unit-tests
+        run: make -j$(($(nproc) + 1)) unit-test
```

Comment:
Isn't simply `-j` enough? Why nproc + 1, did you measure that this is faster?

Reply:
`-j` means parallelize as much as possible, so building all c files at the same time. Usually it is recommended to use 1 + number of CPUs to get max overall speed

Reply:
There seems to have been some discussion recently to use fewer parallel jobs on systems with high cpu/memory ratio:

https://unix.stackexchange.com/questions/208568/how-to-determine-the-maximum-number-to-pass-to-make-j-option

But the second answer still mentions the nrpoc+1, which is what I historically have used.